src/Bicep.Core.IntegrationTests/ParamsParserTests.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.Parsing;
+using Bicep.Core.Samples;
+using Bicep.Core.Syntax;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+using System;
+
+namespace Bicep.Core.IntegrationTests
+{
+    [TestClass]
+    public class ParamsParserTests
+    {
+        private class SyntaxCollectorVisitor : SyntaxVisitor
```

Comment:
This is the same as the syntax collector visitor in PaserTests. It might be worth extracting it to a new file and make it public helper class.